src/utilities/use-mounted.ts

```diff
@@ -0,0 +1,11 @@
+import {useState, useEffect} from 'react';
+
+export function useMounted() {
```

Comment:
> It'd be great to add a docblock describing what the hook does and returns (like useUniqueId

Yes, 😍 I've always wished we used jsdocs for our utilities.

`useIsMountedState` and `useIsMountedRef` would have two differently effects. The state version is so we can render empty values on the version then hydrate and immediately render on the client. Which is something we do quite often with dom measurements. Another idea ive been having is the replication of `componentDidMount`. Occasionally we'll have a situation like this.
```
useEffect(() => {
  console.log(someDep)
// eslint-disable-next-line react/exhaustive-deps
}, [])
``` 

Which could be achieved with
```
const hasBeenMounted = useHasBeenMounted()
if (!hasBeenMounted) console.log('componentDidMount')
```

It also wouldn't need cleanup in the effect, since it wouldn't execute, since we're settings state in an unmounting component. Which is similarly the main purpose of `isMountedRef` to preventing `setState` in unmounted components (aka probably memory leaks)

Reply:
So option 2 :) Some name that makes that clearer would be handy as on inital glance  `useMounted` would imply it changes if the component is mounted or not

More name riffing -  What do you think to `isAfterInitialMount` and `<AfterInitialMount>`? it says it'll only change the once and it keeps the present tense isThing convention (that wasEverMounted and hasBeenMounted lack) and also you can keep a similar name with the helper component: